### PR TITLE
Add `--no-console` and `--working-dir` option to `cocos run` command.

### PR DESCRIPTION
**Pull request by xpol**
This will allow cocos command to be integrated with IEDs.
The IDE runner can only parse output form stdout and stderr.
By disable the console window the game executable will prints to stdout.

### CODE REVIEW

**Comment by xpol** [comment]
@minggo @zilongshanren 

Would you please help merge this?

I'm writing a build plugin for Atom build. With this PR, Atom can tun into a nice "IDE" for cocos2d-x develop.

And the [`build-cocos`](https://atom.io/packages/build-cocos) is published now.

**Comment by darkdukey** [comment]
This is pretty awesome

**Comment by xpol** [comment]
@pandamicro 
Would you please consider merge this?
Thank you very much!

File: bin/strings.json [hunk 1]
```diff
@@ -49,6 +49,8 @@
         "RUN_ARG_MODE" : "Set the run mode, should be debug|release, default is debug.",
         "RUN_ARG_GROUP_WEB" : "web project arguments",
         "RUN_ARG_BROWSER" : "Specify the browser to open the url. Use the system default browser if not specified.",
+        "RUN_ARG_NO_CONSOLE" : "Disable console window (by passing '-console no') for Windows, Mac and Linux.",
```
Comment (natural-law) [inline bin/strings.json]: `'-console no'` should be `'--no-console'`

File: bin/strings.json [hunk 2]
```diff
@@ -387,6 +389,8 @@
         "RUN_ARG_MODE" : "设置运行模式，可选值为 debug/release，默认值为 debug。",
         "RUN_ARG_GROUP_WEB" : "web 工程可用参数",
         "RUN_ARG_BROWSER" : "设置指定浏览器打开 url。如果未指定使用系统默认浏览器。",
+        "RUN_ARG_NO_CONSOLE" : "关闭 Windows，Mac 和 Linux 下的控制台窗口（传入参数'-console no'）。",
```
Comment (natural-law) [inline bin/strings.json]: `'-console no'` should be `'--no-console'`

File: bin/strings.json [hunk 3]
```diff
@@ -724,6 +728,8 @@
         "RUN_ARG_MODE" : "設置運行模式，可選值為 debug/release，默認值為 debug。",
         "RUN_ARG_GROUP_WEB" : "web 工程可用參數",
         "RUN_ARG_BROWSER" : "設置指定流覽器打開 url。如果未指定使用系統默認流覽器。",
+        "RUN_ARG_NO_CONSOLE" : "關閉 Windows，Mac 和 Linux 下的控制檯視窗（傳入'-console no'）。",
```
Comment (natural-law) [inline bin/strings.json]: `'-console no'` should be `'--no-console'`

**Comment by natural-law** [comment]
@xpol It's a really awesome PR. Please solve the typo in `strings.json`. Then I will merge it. Thank you!

**Comment by xpol** [comment]
@natural-law

That's not a typo, It pass `-console no` to cocos _simulator_ see: https://github.com/cocos2d/cocos2d-x/blob/v3/tools/simulator/libsimulator/lib/ProjectConfig/ProjectConfig.cpp#L342

May be I can describe it like: `"关闭 Windows，Mac 和 Linux 下的模拟器控制台窗口（传入参数 '-console no' 给模拟器）。",`

**Comment by xpol** [comment]
@natural-law 
Would you please consider merge #329 ? That is a simple but useful.

**Comment by natural-law** [comment]
> That's not a typo, It pass -console no to cocos simulator see: https://github.com/cocos2d/cocos2d-x/blob/v3/tools/simulator/libsimulator/lib/ProjectConfig/ProjectConfig.cpp#L342
> 
> May be I can describe it like: "关闭 Windows，Mac 和 Linux 下的模拟器控制台窗口（传入参数 '-console no' 给模拟器）。",

I'm sorry about my misunderstanding of the PR.

If the PR is for `the projects which are using libsimulator`. What will be happen if the `-console no` & `-workdir <path>` passed to a normal project? 

For example : If I run a command `cocos run --no-conosle --working-dir <path>` for a project which is not using `libsimulator`, what's the result? Is the project running correctly?

**Comment by xpol** [comment]
Sorry I'm not have all project type experence of cocos (I just use Lua projects).
Is there any projects that init with `cocos new ....` and it does not using `libsimulator` on Windows/Mac/Linux?

**Comment by natural-law** [comment]
@xpol 
The project `tests/cpp-empty-test` is not using `libsimulator`. You can take a test on it. Thanks.
I'm trying it now.

**Comment by xpol** [comment]
No, this PR does not support projects without libsimulator.

**Comment by natural-law** [comment]
> No, this PR does not support projects without libsimulator.

What's your mean of `not support`? 
- If it's just no any effect, I think is OK enough. You should add more description about the new parameters. For example: it's only support projects with libsimulator.
- If the projects without libsimulator are not running correctly. I'm sorry that I can not merge the PR.

**Comment by xpol** [comment]
just no any effect, becasue `-workdir` and `-console` are parsed in `libsimulator`:

https://github.com/cocos2d/cocos2d-x/blob/v3/tools/simulator/libsimulator/lib/ProjectConfig/ProjectConfig.cpp#L283
https://github.com/cocos2d/cocos2d-x/blob/v3/tools/simulator/libsimulator/lib/ProjectConfig/ProjectConfig.cpp#L342

**Comment by natural-law** [comment]
Thanks for your contribution!

**Pull request merged**